backend/src/common/server-application/plugins/authorization/authorization.plugin.ts

```diff
@@ -0,0 +1,32 @@
+import { type FastifyPluginCallback } from 'fastify';
+import fp from 'fastify-plugin';
+
+import { HttpCode } from '~/common/http/http.js';
+
+type AuthOptions = {
+    routesWhiteList: string[];
+};
+
+const authorizationPlugin: FastifyPluginCallback<AuthOptions> = (
+    fastify,
+    { routesWhiteList },
+) => {
+    fastify.addHook('onRequest', (request, reply) => {
```

Comment:
Create special enums like ControllerHook which has ON_REQUEST = 'onRequest' and use this enums in this hook